src/algebra/pi_instances.lean

```diff
@@ -154,8 +154,135 @@ instance is_ring_hom_pi
   is_ring_hom (λ x b, f b x) :=
 (show γ →+* Π a, β a, from pi.ring_hom (λ a, ring_hom.of (f a))).is_ring_hom
 
+-- Note that we only define `single` here for dependent functions with additive fibres.
+section
+variables [decidable_eq I]
+variables [Π i, has_zero (f i)]
+
+/-- The function supported at `i`, with value `x` there. -/
+def single (i : I) (x : f i) : Π i, f i :=
+λ i', if h : i' = i then (by { subst h, exact x }) else 0
+
+@[simp]
+lemma single_eq_same (i : I) (x : f i) : single i x i = x :=
+begin
+  dsimp [single],
+  split_ifs,
+  { refl, },
+  { exfalso, exact h rfl, }
+end
+@[simp]
+lemma single_eq_of_ne {i i' : I} (h : i' ≠ i) (x : f i) : single i x i' = 0 :=
+begin
+  dsimp [single],
+  split_ifs with h',
+  { exfalso, exact h h', },
+  { refl, }
+end
+
+end
+
 end pi
 
+section
+universes u v
+variable {I : Type u}     -- The indexing type
+variable (f : I → Type v) -- The family of types already equipped with instances
+variables [Π i, monoid (f i)]
+
+/-- Evaluation of functions into an indexed collection of monoids at a point is a monoid homomorphism. -/
+@[to_additive "Evaluation of functions into an indexed collection of additive monoids at a point is an additive monoid homomorphism."]
+def monoid_hom.apply (i : I) : (Π i, f i) →* f i :=
```

Comment:
Should there also be a `ring_hom` version of this one?

Reply:
Done. Also added some `ring` analogues below.